WebApi/WebApiTest/ServiceTest/CommentServiceTest.cs

```diff
@@ -0,0 +1,216 @@
+using WebApi.Models.DTO;
+using WebApi.Models.POCO;
+using WebApi.Services.Serives_Implementations;
+using Xunit;
+
+namespace WebApiTest.ServiceTest
+{
+    public class CommentServiceTest
+    {
+        const int UserId = 1;
+        List<Comment> comments = new List<Comment>
+        {
+            new Comment() { CommentID = 1, UserID = 1, PostID = 1, DateTime = DateTime.Today, Content = "test" },
+             new Comment() { CommentID = 2, UserID = 2, PostID = 2, DateTime = DateTime.Today, Content = "test2" },
+            new Comment() { CommentID = 3, UserID = 3, PostID = 3, DateTime = DateTime.Today, Content = "test3" }
+
+        };
+
+        [Fact]
+        public void GetById_ValidCall()
+        {
+            int expectedID = 1;
+            var expected = comments.Where(x => x.CommentID == expectedID).FirstOrDefault();
+            var mockICommentRepository = new Mock<ICommentRepository>();
+            mockICommentRepository.Setup(x => x.GetById(expectedID)).Returns(expected);
+
+            var commentService = new CommentService(mockICommentRepository.Object);
+            var actual = commentService.GetById(expectedID, UserId);
+
+            Assert.True(actual != null);
+            Assert.Equal(expected.CommentID, actual.CommentID);
+            Assert.Equal(expected.Content, actual.Content);
+            Assert.Equal(expected.DateTime, actual.DateTime);
+            Assert.Equal(expected.PostID, actual.PostID);
+            Assert.Equal(expected.UserID, actual.UserID);
+
+
+        }
+
+        [Fact]
+        public void GetById_InValidCall()
+        {
+            int expectedID = 0;
+            var expected = comments.Where(x => x.CommentID == expectedID).FirstOrDefault();
+            var mockICommentRepository = new Mock<ICommentRepository>();
+            mockICommentRepository.Setup(x => x.GetById(expectedID)).Returns(new Comment { });
+
+            var commentService = new CommentService(mockICommentRepository.Object);
+            var actual = commentService.GetById(expectedID, UserId);
+
+            Assert.True(actual.CommentID == 0);
+            Assert.Null(actual.Content);
+            ;
+
+
+        }
+        [Fact]
+        public void GetAll_ValidCall()
+        {
+            var expected = comments;
+            var mockICommentRepository = new Mock<ICommentRepository>();
+            mockICommentRepository.Setup(x => x.GetAll()).Returns(expected.AsQueryable());
+
+            var commentService = new CommentService(mockICommentRepository.Object);
+            var actual = commentService.GetAll(UserId).ToList();
+
+            Assert.True(actual != null);
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.True(expected.All(item => actual.Any(actualItem => item.CommentID == actualItem.CommentID &&
+              item.UserID == actualItem.UserID && item.Content == actualItem.Content &&
+              item.PostID == actualItem.PostID)));
```

Comment:
Tutaj wystarczy chyba ostatnia linijka porównująca. Nie ma potrzeby walczyć o optymalizację testów, więc moim zdaniem powinniśmy postawić na większą czystość kodu. Nie musimy sztywno trzymać się zasady jednego Assertu, ale jeśli jest taka możliwość to się postarajmy :)